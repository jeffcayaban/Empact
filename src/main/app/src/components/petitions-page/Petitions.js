import React from 'react'

import {Tabs, Tab, Grid, Col} from 'react-bootstrap';
import PetitionsTable from "./PetitionsTable";
import {withRouter} from "react-router-dom";
import {ASCENDING, DESCENDING, LEAST_DISCUSSED, MOST_DISCUSSED, NEWEST, OLDEST} from "../../constants";

/**
 * Displays the petitions for the explore petitions page.
 */

class Petitions extends React.Component {

    constructor(props) {
        super(props);
        this.state = { currentTab: 1 };
        this.handleSelect = this.handleSelect.bind(this);
    }

    // Updates the component's state to reflect the current tab of petitions that is selected.
    handleSelect(key) {
        (key !== this.state.currentTab) && this.setState({ currentTab: key });
    }

    render() {
        const currentTab = this.state.currentTab;
        return (
            <div>
                <div className={"centerStyle"}>
                    <h2>Petitions</h2>
                    <p align="center">Explore the petitions!</p>
                </div>
                <div className={'secondaryPageContainerStyle'}>
                    <Grid className={"divNoPadding"} style={{ paddingTop: '30px' }}>
                        <Col md={12} lg={12} className={"divNoPadding"}>
                            <Tabs id={"petitions-table-tabs"} activeKey={currentTab} onSelect={this.handleSelect}>
                                <Tab eventKey={1} title={NEWEST}>
                                    { currentTab === 1 && <PetitionsTable sort={DESCENDING} byNoOfArgs={false}/> }
                                </Tab>
                                <Tab eventKey={2} title={OLDEST}>
                                    { currentTab === 2 && <PetitionsTable sort={ASCENDING} byNoOfArgs={false}/> }
                                </Tab>
                                <Tab eventKey={3} title={MOST_DISCUSSED}>
                                    { currentTab === 3 && <PetitionsTable sort={DESCENDING} byNoOfArgs={true}/> }
                                </Tab>
                                <Tab eventKey={4} title={LEAST_DISCUSSED}>
                                    { currentTab === 4 && <PetitionsTable sort={ASCENDING} byNoOfArgs={true}/> }
                                </Tab>
                            </Tabs>
                        </Col>
                    </Grid>
                </div>
            </div>
        )
    }
}

export default withRouter(Petitions);