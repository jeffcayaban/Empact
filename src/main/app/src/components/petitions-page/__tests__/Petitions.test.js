import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import Petitions from "../Petitions";

describe('PetitionsTable', () => {

    it('should render successfully and get the most discussed petitions', () => {
        const component = shallow(<Petitions.WrappedComponent params={{ router: jest.fn() }} />);
        expect(toJson(component)).toMatchSnapshot();
    });

});
