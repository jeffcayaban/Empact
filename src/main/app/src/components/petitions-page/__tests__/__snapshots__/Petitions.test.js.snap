// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetitionsTable should render successfully and get the most discussed petitions 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Petitions
    </h2>
    <p
      align="center"
    >
      Explore the petitions!
    </p>
  </div>
  <div
    className="secondaryPageContainerStyle"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
      style={
        Object {
          "paddingTop": "30px",
        }
      }
    >
      <Col
        bsClass="col"
        className="divNoPadding"
        componentClass="div"
        lg={12}
        md={12}
      >
        <Uncontrolled(Tabs)
          activeKey={1}
          id="petitions-table-tabs"
          onSelect={[Function]}
        >
          <Tab
            eventKey={1}
            title="Newest"
          >
            <PetitionsTable
              byNoOfArgs={false}
              sort="desc"
            />
          </Tab>
          <Tab
            eventKey={2}
            title="Oldest"
          />
          <Tab
            eventKey={3}
            title="Most Discussed"
          />
          <Tab
            eventKey={4}
            title="Least Discussed"
          />
        </Uncontrolled(Tabs)>
      </Col>
    </Grid>
  </div>
</div>
`;
